scripts/dex-router/src/index.ts

```diff
@@ -178,48 +198,54 @@ export class DexRouter {
         ixsFlashTakeOrder.endFlashIx,
         ...ixsFlashTakeOrder.closeWsolAtaIxs,
       ],
-      routeBest.lookupTableAddresses ?? []
+      route.lookupTableAddresses ?? []
     );
     tx.sign([this.executor]);
-
-    return {
-      transaction: Buffer.from(tx.serialize()).toString("base64"),
-      chain_id: this.chainId,
-    };
+    return tx;
   }
 
   /**
-   * Examines routes generated by all available routers and returns the one that yields the most output amount
-   * @param inputMint The mint of the token to be sold through the router
-   * @param outputMint The mint of the token to be bought through the router
-   * @param amountIn The amount of the input token to be sold
+   * Examines routes generated by all available routers and returns the one that yields the most output amount. Filters out routes that exceed the max tx size.
+   * @param order The order to be fulfilled
+   * @returns The best route and the transaction that fulfills the order
    */
   private async getBestRoute(
-    inputMint: PublicKey,
-    outputMint: PublicKey,
-    amountIn: bigint
-  ): Promise<RouterOutput> {
-    const routerOutputs = (
+    order: OrderStateAndAddress
+  ): Promise<RouterOutputAndTx> {
+    const routerInfos = (
       await Promise.all(
         this.routers.map(async (router) => {
           try {
             const routerOutput = await router.route(
-              inputMint,
-              outputMint,
-              amountIn
+              order.state.inputMint,
+              order.state.outputMint,
+              order.state.remainingInputAmount
+            );
+            const routerTx = await this.createRouterTransaction(
+              routerOutput,
+              order
             );
-            return routerOutput;
+            return { output: routerOutput, tx: routerTx };
           } catch (error) {
             console.error(`Failed to route order: ${error}`);
           }
         })
       )
-    ).filter((routerOutput) => routerOutput !== undefined);
-    if (routerOutputs.length === 0) {
+    )
+      .filter((routerInfo) => {
+        if (routerInfo === undefined) {
+          return false;
+        }
+        return routerInfo.tx.serialize().length <= 1232;
```

Comment:
define a constant for this

Reply:
I think this is not a good place to handle this size limit. Maybe `createRouterTransaction` should throw an exception if it can not a create tansaction within the size limit or the output of it can be possibly null to signify it wasn't successful.